public/models/GameMap.js

```diff
@@ -294,15 +285,82 @@ class GameMap {
             this.traps.push(newTrap);
             break;
           } // else it falls down to dafault case
+        case 'OnOffTrap':
+          const trapImg = [onOffTrap, trap][ Utilities.getRandomNumber(0, 2)];
```

Comment:
Not in this section exactly but you might have something set up for the coal trap incorrectly? Because it is taking the damage when it should be off.